webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -253,61 +175,334 @@ class ResponsePage extends Component {
+
+                this.setState({
+                    applicationData: updateApplicationData,
+                    removeTagModalVisible: false,
+                    tagToRemove: null
+                });
+            }
+            else {
+                this.cancelRemoveTag();
+                this.error(response.message);
+            }
+        });
+    };
+
+    cancelRemoveTag = () => {
+        this.setState({
+            tagToRemove: null,
+            removeModalVisible: false,
+        });
+    };
 
     renderTags() {
         const data = this.state.applicationData;
+
         if (data) {
-            let tags = data.tags.map(tag => {
+            let tags = data.tags ? data.tags.map(tag => {
                 return <span
                     key={tag.id}
                     className="btn badge badge-info"
                 >
                     {tag.name}
                     <i onClick={(e) => this.removeTag(tag.id)} className="far fa-trash-alt"></i></span>
-            });
+            })
+                :
+                null
             return tags
         };
-    }
+    };
 
     renderTagModal() {
         const { eventLanguages } = this.state;
+
         if (eventLanguages) {
             return <TagModal
+                keys={this.state.keys}
+                i18nt={this.props.i18n}
                 t={this.props.t}
-                postTag={this.addTag}
+                postTag={(tag) => this.addTag(tag)}
                 eventLanguages={eventLanguages}
             />
         };
-    }
+    };
+
+
+    // Reviews
+    // Render Reviews
+    renderReviews() {
+
+        if (this.state.applicationData) {
+            if (this.state.applicationData.reviewers) {
+                const reviews = this.state.applicationData.reviewers.map((val, index) => {
+                    let num = index + 1;
+                    //   {"reviewer_user_id": 4, "user_title": "Mr", "firstname": "Joe", "lastname": "Soap", "completed": false},
+                    if (!val) {
+                        return <div
+                            className="add-reviewer"
+                            key={index}
+                        >
+                            <button
+                                data-toggle="modal"
+                                type="button"
+                                data-target="#exampleModalReview"
+                                className="btn btn-light">
+                                Assign Reviewer
+                                </button>
+                        </div>
+                    }
+                    else {
+                        return <div className="reviewer">
+                            <label>{"Reviewer" + num}</label>
+                            <div>
+                                <p>{val.user_title} {val.firstname} {val.lastname}</p>
+                                
+                                {val.completed ? <p className="review-completed">Completed</p>
```

Comment:
Please translate the text in this section ("Assign Reviewer", "Completed" and "Incomplete")

Reply:
@greeffjean - please fix this.